docs/in-depth/configuration-file.md

```diff
@@ -44,30 +45,51 @@ my_inputs = { temp_vector[1], temp_vector[3] }
+## Declaring input particles
 
-Inputs:
-
-* `input::particles`: Vector of LorentzVectors passed as argument to `computeWeights()`
-
-* `input::met`: LorentzVector of the observed MET (optional argument to `computeWeights()`)
-
-* `cuba::ps_points`: Vector of doubles between 0 and 1. Phase-space point, ie. set of random numbers launched by Cuba to perform the integration. The length of the vector is the total number of integration dimensions declared by the modules. Instead of manually specifying the `cuba::ps_points/i` input tags, it is possible to use to `getpspoint()` function (which automatically increases “i” each time it is called).
-
-* `cuba::ps_weight`: (double) Weight associated with the phase-space point
-
-Outputs:
+Particles passed as input to the `computeWeights()` function (see [Calling MoMEMta](calling-momemta)) can be linked with the modules by registering them in the Lua configuration. For instance, for a particle declared in C++ as `momemta::Particle m_part1 { "part1", p1, 0 }`, one would do:
+```Lua
+local l_part1 = declare_input("part1") -- Pass as argument the name string of m_part1
+```
+The Lua object `l_part1` now contains the input tag needed to access the particle's 4-vector. For instance, defining a transfer function on the energy of `part1` can be done as:
+```Lua
+GaussianTransferFunctionOnEnergy.tf_part1 = {
+  ps_point = add_dimension(), -- A transfer function integrates over a variable (the particle's energy), so we need a new dimension in the integrated volume
+  reco_particle = l_part1.reco_p4, -- Pass the input tag corresponding to the experimentally reconstructed 4-vector of the particle, passed to 'computeWeights()'
+  sigma = 0.10 -- Take 10% resolution on the energy
+}
+```
+Since the transfer function integrates over the particle's energy, it generates new values for its energy. This defines a new, 'parton'-level 4-vector for `part1`, which can now be passed e.g. to a matrix element. In this example, the parton-level 4-vector can be accessed directly through the `tf_part1::output` input tag. It is also possible to register it with the `l_part1` object:
+```Lua
+l_part1.set_gen_p4("tf_part1::output")
+```
+When declaring further modules, instead of using `tf_part1::output`, it is now possible to use `l_part1.gen_p4` to access the parton-level 4-vector, i.e. these are equivalent:
+```Lua
+P4Printer.printer1 = { input = "tf_part1::output" }
+P4Printer.printer2 = { input = l_part1.gen_p4 }
+```
 
-* `integrands`: Vector of doubles. This is the value taken to be the integrand, returned to Cuba to compute the integral. Exactly **one** declared module **must** produce this quantity.
+Finally, the missing transverse energy 4-vector (passed optionally to `computeWeights()`) can either be accessed directly through the `met::p4` input tag, or declared as a Lua object as above:
+```Lua
+local met = declare_input("met")
+```
```

Comment:
I did not test, but I'm prettry sure this won't work. The `met` input is always declared, you don't need to do anything to access it.